include/OpenSpaceToolkit/Astrodynamics/Estimator/ODLeastSquaresSolver.hpp

```diff
@@ -0,0 +1,138 @@
+using ostk::mathematics::object::VectorXd;
+
+using ostk::astrodynamics::solver::LeastSquaresSolver;
+using ostk::astrodynamics::trajectory::LocalOrbitalFrameFactory;
+using ostk::astrodynamics::trajectory::NumericalSolver;
+using ostk::astrodynamics::trajectory::Orbit;
+using ostk::astrodynamics::trajectory::Propagator;
+using ostk::astrodynamics::trajectory::State;
+using ostk::astrodynamics::trajectory::state::CoordinateSubset;
+
+/// @brief Orbit Determination solver using least squares
+class ODLeastSquaresSolver
+{
+   public:
+    class Analysis
+    {
+       public:
+        /// @brief Constructor
+        Analysis(const State& aDeterminedState, const LeastSquaresSolver::Analysis& anAnalysis);
+
+        /// @brief Destructor
+        ~Analysis() = default;
+
+        /// @brief Print analysis
+        friend std::ostream& operator<<(std::ostream& anOutputStream, const Analysis& anAnalysis);
+
+        /// @brief Print analysis
+        void print(std::ostream& anOutputStream) const;
+
+        /// @brief Get determined state
+        const State& getDeterminedState() const;
+
+        /// @brief Get analysis
+        const LeastSquaresSolver::Analysis& getSolverAnalysis() const;
+
+       private:
+        State determinedState_;
+        LeastSquaresSolver::Analysis solverAnalysis_;
+    };
+
+    static const Shared<const Frame> EstimationFrameSPtr;
+
+    /// @brief Constructor
+    ///
+    /// @param anEnvironment Environment
+    /// @param aNumericalSolver Numerical solver
+    /// @param aSolver Least squares solver
+    ODLeastSquaresSolver(
+        const Environment& anEnvironment,
+        const NumericalSolver& aNumericalSolver = NumericalSolver::Default(),
+        const LeastSquaresSolver& aSolver = LeastSquaresSolver::Default()
+    );
+
+    /// @brief Access environment
+    const Environment& accessEnvironment() const;
+
+    /// @brief Access propagator configuration
+    const Propagator& accessPropagator() const;
+
+    /// @brief Access solver
+    const LeastSquaresSolver& accessSolver() const;
+
+    /// @brief Estimate state using least squares
+    ///
+    /// @param anInitialGuessState Initial guess state
+    /// @param aReferenceStateArray Reference states to minimize error against
+    /// @param anInitialGuessSigmas Initial guess sigmas
+    /// @param aReferenceStateSigmas Reference state sigmas
+    /// @param anEstimationCoordinateSubsets Coordinate subsets to estimate
+    Analysis estimateState(
+        const State& anInitialGuessState,
+        const Array<State>& aReferenceStateArray,
+        const Array<Shared<const CoordinateSubset>>& anEstimationCoordinateSubsets =
+            Array<Shared<const CoordinateSubset>>::Empty(),
+        const std::unordered_map<CoordinateSubset, VectorXd>& anInitialGuessSigmas = DEFAULT_INITIAL_GUESS_SIGMAS,
+        const std::unordered_map<CoordinateSubset, VectorXd>& aReferenceStateSigmas = DEFAULT_REFERENCE_STATE_SIGMAS
+    ) const;
+
```

Comment:
Missing compute state and orbit residuals method

Reply:
As discussed these will be added later due to the complexity of the signatures